### PR TITLE
fix(sdk-core): fix pending approval for consolidation

### PR DESCRIPTION
**Pull request by davidkaplanbitgo**
When doing an unspent consolidation, there are no recipients in the `buildParams` because they are generated within WP. This causes errors reconstructing the consolidation transaction during pending approval. This new check sets the `canRecreateTransaction=false` if there are no recipients. Instead, it will use the txHex of the transaction passed into `/send` and use that. 

This should unblock customers who have policies on BTC wallets that want to consolidate.

BG-78492

### CODE REVIEW

File: modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts [hunk 1]
```diff
@@ -178,6 +178,11 @@ export class PendingApproval implements IPendingApproval {
       canRecreateTransaction = false;
     }
 
+    // If there are no recipients, then the transaction cannot be recreated
+    if (!!this.info()?.transactionRequest?.buildParams?.recipients) {
+      canRecreateTransaction = false;
+    }
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts:184]: Can we add a test for this?

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts [hunk 2]
```diff
@@ -178,6 +178,11 @@ export class PendingApproval implements IPendingApproval {
       canRecreateTransaction = false;
     }
 
+    // If there are no recipients, then the transaction cannot be recreated
+    if (!!this.info()?.transactionRequest?.buildParams?.recipients) {
```
Comment (brandonblack) [inline modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts:182]: I think this should be `if (!this.info()?.transactionRequest?.buildParams?.recipients?.length && this.info()?.transactionRequest?.buildParams?.type !== 'consolidate') {`

Or possibly nicer:
```
const recipients = this.info()?.transactionRequest?.buildParams?.recipients || [];
const type = this.info()?.transactionRequest?.buildParams?.type;
if (recipients.length === 0 && type !== 'consolidate') {
```

**Review by brandonblack — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts (hunk 1 above)
Comment (davidkaplanbitgo) [reply modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts]: Added in latest push

**Review by brandonblack — APPROVED** [review]

**Review by angshal — APPROVED** [review]

File: modules/bitgo/test/v2/unit/pendingApproval.ts [hunk 3]
```diff
@@ -89,6 +89,51 @@ describe('Pending Approvals:', () => {
     scope.done();
   });
 
+  it('should no recreate the transaction during approving a pending approval if there are no recipients', async () => {
```
Comment (zahin-mohammad) [inline modules/bitgo/test/v2/unit/pendingApproval.ts:92]:
```suggestion
  it('should not recreate the transaction during approving a pending approval if there are no recipients', async () => {
```

**Review by zahin-mohammad — APPROVED** [review]

**Pull request merged**